src/tests/comparison_tests/test_s21_is_less.c

```diff
@@ -1,47 +1,56 @@
 #include "../s21_decimal_test.h"
 
 START_TEST(gmp_random) {
-    s21_decimal a1 = {0};
-    init_zero(&a1);
-
+    s21_decimal dec1 = {0};
     mpz_t mpz_val, mpz_copy;
-    mpz_init(mpz_val);
-    mpz_init(mpz_copy);
-    mpz_set_ui(mpz_val, 0);
-    mpz_set_ui(mpz_copy, 0);
 
-    get_random_pair(&a1, &mpz_val, 3);
+    mpz_init_set_ui(mpz_val, 0);
+    mpz_init_set_ui(mpz_copy, 0);
 
-    convert_decimal_to_gmp(&mpz_copy, &a1);
-    s21_decimal res = convert_gmp_to_decimal(mpz_val);
+    get_random_pair(&dec1, &mpz_val, 3);
 
-    /* Produces randomly smaller / greater numbers */
-    if (rand() % 2)
-        set_exponent(&a1, (get_exponent(a1) + 1) % 28);
+    convert_decimal_to_gmp(&mpz_copy, &dec1);
+    s21_decimal dec2 = convert_gmp_to_decimal(mpz_val);
 
-    /* (!) Hack. Change this to our own function */
-    tmp_normalize_exponent(&a1);
+    /* d1 is always < d2 */
+    gmp_sum_int(&mpz_copy, 100);
 
     // Compare op1 and op2. Return a positive value if op1 > op2, zero if op1 = op2, or a negative value if op1 < op2.
-    int got = (s21_is_less(a1, res));
-    int expected = mpz_cmp(mpz_val, mpz_copy);
+    int got = s21_is_less(dec1, dec2);
 
-    mpz_clear(mpz_copy);
-    mpz_clear(mpz_val);
+    /* BUG (!) MPZ IS UNABLE TO CORRECTLY COMPARE UNSIGNED INTS */
+    int expected = mpz_cmp(mpz_val, mpz_copy);
```

Comment:
@bezlant please, draw your attention to this line. 

There is one more problem that is generally quite easily fixable. All our GMPs are unsigned. I decided to use uints, because they do not have sign bit. GMP ints are completely incompatible with s21_decimal binary representation, as sign bit breaks everything.

GMP uints can actually be signed & it works quite well, but, unfortunately, GMP incorrectly compares negative uints. Thus, we need to write custom wrapper around this mpz function. I created special module for GMP wrappers (see helpers -> GMP directory). Something like this is also needed for s21_add. 

In this wrapper we will check signs of GMPs & then compare their absolute values.